Feedback Option

a-augustin opened the pull request:
## Description ✏️
I added the feedback page in the submodule. 

## Checklist ✅
![Screen Shot 2024-10-28 at 12 18 46 AM](https://github.com/user-attachments/assets/2dae5422-8f9d-4b83-bdac-d7f0f511ae36)
![Screen Shot 2024-10-28 at 12 19 02 AM](https://github.com/user-attachments/assets/5f7c0bb6-2607-43a0-97ef-d9aa35f83303)


- [ x] I have added screenshots (if UI changes are present).
- [ x] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).

a-augustin committed:
added feedback page and placeholder for api endpoint

nang-dev commented:
Ohhh sorry augustin. we should have it be automatic! so theyre already signed in. client side (the app) can just send their api key and the feedback as text. so just one box -> feedback and one button -> send. and we should have auto close, with a vscode notif in the bottom right that says "Feedback sent successfully!" (or error)

so basically:

1. see if user is signed in (look at the pearai credentials), if not, ask them to sign in to submit feedback
2. send headers (https://github.com/trypear/pearai-submodule/blob/main/core/continueServer/stubs/headers.ts#L4
), api-key (see PearAIServer.ts), and feedback to SERVER_URL/client_feedback


my bad for not making this clear earlier

a-augustin commented:
No worries! I’ll make the changes. Thanks for the feedback

a-augustin committed:
updated feedback option

a-augustin commented:
I updated the feedback page with just the text box and the vscode notifications when the user submits. I also added the headers and API key to the endpoint you mentioned. If there are any issues, let me know.

Himanshu-Singh-Chauhan commented on gui/src/App.tsx:
```diff
@@ -19,106 +19,85 @@ import LocalOnboarding from "./pages/onboarding/LocalOnboarding";
+const router = createMemoryRouter([
   {
-    initialEntries: [window.isPearOverlay ? "/inventory" : window.initialRoute], 
+    path: "/",
```
why is all this removed?

Himanshu-Singh-Chauhan commented on gui/src/App.tsx:
```diff
@@ -19,106 +19,85 @@ import LocalOnboarding from "./pages/onboarding/LocalOnboarding";
+      {
+        path: "/feedback",
+        element: <Feedback/>,
+      },
```
you can just add /feedback route here, leave the rest as it is 😄

Himanshu-Singh-Chauhan commented on gui/src/components/Layout.tsx:
```diff
@@ -118,10 +118,6 @@ const HIDE_FOOTER_ON_PAGES = [
-  "/inventory",
-  "/inventory/aiderMode",
-  "/inventory/perplexityMode"
 ];
```
dont remove these, we want to hide footers on those routes

a-augustin replied:
I didn't remove those footer line, I'm not sure how they were removed.

a-augustin committed:
Revert "updated feedback option"

This reverts commit e7ae292af134cd9d9101a017b21fe575fe2ed062.

a-augustin committed:
fixed merge conflicts

a-augustin committed:
updated feedback changes

charlwillia6 commented:
Whatever happened with this?  Are you guys still looking to have a feedback option?
